tensorboard/plugins/histogram/summary_v2.py

```diff
@@ -427,7 +427,11 @@ def _buckets_v3(data, bucket_count=None):
         data = tf.reshape(data, shape=[-1])  # flatten
         data = tf.cast(data, tf.float64)
         data_size = tf.size(input=data)
-        is_empty = tf.equal(data_size, 0)
+        # If bucket_count is specified as zero, an empty tensor of shape
```

Comment:
So based on the design spec, there is actually a distinction in these two cases, where "empty" in terms of no buckets should be the `(0, 3)` tensor, but "empty" in terms of no data is supposed be the `(num_buckets, 3)` tensor of all zero values.  (This is a change from the v2 behavior of also returning the empty `(0, 3)` tensor in that case, which means that the `test_empty_input` tests in the base case would need overrides for the v3 format.)  It's ok to not implement that part in this PR, we could do it in a later one, but just wanted to highlight that in case it wasn't clear.

That said, the two cases _can_ still both be handled as one condition, like you have it here, if we change the `when_empty()` implementation to:

```
def when_empty():
    return tf.zeros((bucket_count, 3), dtype=tf.float64)
```

That will naturally handle the case of 0 buckets by emitting the empty tensor of shape (0, 3).